autometa/common/exceptions.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Copyright 2020 Ian J. Miller, Evan R. Rees, Kyle Wolf, Siddharth Uppal,
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+
+File containing customized AutometaExceptions for more specific exception handling
+"""
+
+class AutometaException(Exception):
+    """docstring for AutometaException."""
+
+    def __init__(self, value):
+        self.value = value
+
+    issue_request = '''
+    An error was encountered!
+
+    Please help us fix your problem!
+
+    You may file an issue with us at https://github.com/KwanLab/Autometa/issues/new
+    '''
+    def __str__(self):
+        return f'{self.value}\n\n{issue_request}'
```

Comment:
It seems that in Autometa.py the whole of `main` is in a try/except block, which also prints this message. So if this error is raised, will the user see the message twice?

Reply:
> You said that Prodigal doesn't support `.gz` files, but could the file be piped into Prodigal with the unix command `zcat`? I'm thinking of something like: `zcat asm.fasta.gz > prodigal ...`.

This is a feature request for the next version of prodigal, however, it does not appear this is currently available. See discussion [here](https://github.com/hyattpd/Prodigal/issues/63)

Similarly you can try:

`gzip tests/data/metagenome.fna && gzip -c tests/data/metagenome.fna.gz | prodigal -m -o /dev/null -p meta -a test.faa -d test.fna`

This results in genes found in sequences with runs of N's and the resulting files are binary encoded.

A few example output lines from test.fna below:

```shell
>^W<82>^Vˈ<91>.<FD><92>0^\<97><9F>~<EA>Ô<C4><F9>^N^YZ<F4>^^}<E0>_1 # 1 # 5538 # 1 # ID=2_1;partial=11;start_type=Edge;rbs_motif=None;rbs_spacer=None;gc_cont=0.930
NNNGTNNNNNNNNCNNNGNNATNGNGNTNNNNNATNANNNNGNNNNNNNNNNNNNNANNNNNGNNNNNNN
NNNNNNCNNNNNNNNNNNNNNNNNNTNNNANNNNNNNNNNANNCNNGNGNNNNNNNNNNNNNNNNNCNNN
NNNNNNNNNNANNNNNNTNNNNGNNNNNNNNNNNNNNNCNNNNNNNNNCNNNNNNNNNTGNNNNNNNNNN
TNNNNNNNNNNGNNNNNNANNNNNNNNNNNTNNNNGTNNNNNNNNNNNNNNNNNNNNNCNNNNNNNNNNN
NNNANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCNNNNGNNNNNNNNNNNNNNNNNNNNNNNNNN
NTNGNNNNNNNCNNNTNCNNNNNNTNNNNNNNGNANANNNANNTTCNNNNNGNNNNGNNNNNNNNANNNN
NNNNNNNNNNNNNNNNNNNNNNNNNNGNNNNANNNACCNNNNNNNNNNNNCNNNNNNNNNCNNNNNNNNN
```